theme/json.go

```diff
@@ -122,10 +140,26 @@ func (t *jsonTheme) Color(name fyne.ThemeColorName, variant fyne.ThemeVariant) c
 }
 
 func (t *jsonTheme) Font(style fyne.TextStyle) fyne.Resource {
+	if val, ok := t.data.Fonts[styleString(style)]; ok {
+		r, err := val.resource()
+		if err != nil {
+			fyne.LogError("Failed to parse URI", err)
+		} else {
+			return r
+		}
+	}
 	return t.fallback.Font(style)
 }
 
 func (t *jsonTheme) Icon(name fyne.ThemeIconName) fyne.Resource {
+	if val, ok := t.data.Icons[string(name)]; ok {
+		r, err := val.resource()
+		if err != nil {
+			fyne.LogError("Failed to parse URI", err)
+		} else {
+			return r
+		}
```

Comment:
This is exactly the same as in `Font(…)`. Maybe it should be moved into `stringUri.resource()` which then would simply return `nil` if the resource is not present for whatever reason?